packages/block-library/src/post-featured-image/dimension-controls.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * WordPress dependencies
+ */
+import { __, _x } from '@wordpress/i18n';
+import {
+	PanelBody,
+	__experimentalUnitControl as UnitControl,
+	BaseControl,
+	Flex,
+	FlexItem,
+	__experimentalSegmentedControl as SegmentedControl,
+	__experimentalSegmentedControlOption as SegmentedControlOption,
+	__experimentalUseCustomUnits as useCustomUnits,
+} from '@wordpress/components';
+
+const SCALE_OPTIONS = (
+	<>
+		<SegmentedControlOption
+			value="cover"
+			label={ _x( 'Cover', 'Scale option for Image dimension control' ) }
+		/>
+		<SegmentedControlOption
+			value="contain"
+			label={ _x(
+				'Contain',
+				'Scale option for Image dimension control'
+			) }
+		/>
+		<SegmentedControlOption
+			value="fill"
+			label={ _x(
+				'Stretch',
+				'Scale option for Image dimension control'
+			) }
+		/>
+	</>
+);
+
+const DimensionControls = ( {
+	attributes: { width, height, scale },
+	setAttributes,
+} ) => {
+	const units = useCustomUnits( {
+		availableUnits: [ 'px', '%', 'vw', 'em', 'rem' ],
```

Comment:
Should this inherit user options from `spacing.units`? example: https://github.com/WordPress/gutenberg/blob/8d5a38a6d708a0a5a1a4ed5afc81f2df9528d6e7/packages/block-editor/src/components/letter-spacing-control/index.js#L25

Reply:
I'm not really sure about this, but I've updated it as you have worked more on these. @jorgefilipecosta or @jasmussen  could chime in as well..